File: sdk-go/documentutils.go
package llmsdk

// [internal] For providers that do not support document parts,
// we translate them to compatible parts such as Text.
// Inner Document parts are flattened.
func GetCompatiblePartsWithoutDocumentParts(parts []Part) []Part {
	var result []Part
	for _, part := range parts {
		if part.DocumentPart != nil {
			result = append(result, GetCompatiblePartsWithoutDocumentParts(part.DocumentPart.Content)...)
		} else {
			result = append(result, part)
		}
	}
	return result
}

File: sdk-js/src/document.utils.ts
import type { Part } from "./types.ts";

/**
 * For providers that do not support document parts,
 * we translate them to compatible parts such as Text.
 * Inner Document parts are flattened.
 */
export function getCompatiblePartsWithoutDocumentParts(parts: Part[]): Part[] {
  return parts
    .map((part) => {
      if (part.type === "document") {
        return getCompatiblePartsWithoutDocumentParts(part.content);
      }
      return part;
    })
    .flat();
}
